proposals/lip-0044.md

```diff
@@ -190,234 +216,320 @@ genesisDataSchema = {
 
 The genesis data substore stores the timestamp of the genesis block.
 
-### Commands
+### Events
 
-The Validators module does not specify any commands.
+#### GeneratorKeyRegistration
 
-### Protocol Logic for Other Modules
+This event has `typeID = TYPE_ID_GENERATOR_KEY_REGISTRATION`.
+This event is emitted when a generator key is registered.
+The event data contains the generator key `generatorKey` and the result of the registration `result`. 
 
-#### registerValidatorKeys
+##### Topics
 
-This function creates a new validator account in the validators data substore. It is called as part of the [delegate registration][lip-0057] and [authority registration][lip-0047] commands that are part of the DPoS and PoA module.
+* `address`: The address for which the generator key has been registered.
+
+##### Data
+
+```java
+generatorKeyRegDataSchema = {
+    "type": "object",
+    "required": ["generatorKey", "result"],
+    "properties": {
+        "generatorKey": {
+            "dataType": "bytes",
+            "length" : ED25519_PUBLIC_KEY_LENGTH,
+            "fieldNumber": 1
+        },
+        "result": {
+            "dataType": "uint32",
+            "fieldNumber": 2
+        }
+    }
+}
+```
 
-##### Parameters
+#### BLSKeyRegistration
 
-This function has the following input parameters in the order given below:
+This event has `typeID = TYPE_ID_BLS_KEY_REGISTRATION`.
+This event is emitted when a BLS key is registered.
+The event data contains the BLS key `blsKey` and the result of the registration `result`.
 
-* `validatorAddress`: A 20 bytes value identifying the validator account in the validators data substore.
-* `proofOfPossession`: The [proof of possession](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0038.md#public-key-registration-and-proof-of-possession) for `blsKey`.
-* `generatorKey`: A 32 bytes value, specifying the generator key of the validator.
-* `blsKey`: A 48 bytes value, specifying the BLS key of the validator.
+##### Topics
 
-##### Returns
+* `address`: The address for which the BLS key has been registered.
 
-* `success`: A boolean, indicating success or failure of the registration.
+##### Data
 
-##### Execution
+```java
+blsKeyRegDataSchema = {
+    "type": "object",
+    "required": ["blsKey", "result"],
+    "properties": {
+        "blsKey": {
```

Comment:
What is the reason for having the key, but not the PoP here?

Reply:
true, it can be added for completness